tests/plugins/clipboard.test.ts

```diff
@@ -190,16 +192,140 @@ describe("clipboard", () => {
+    expect(getCellContent(model, "B2")).toBe("b2");
+    setStyle(model, "B2", { bold: true });
+    expect(getCell(model, "B2")!.style).toEqual({ bold: true });
+
+    copy(model, "B2");
+
+    setStyle(model, "B2", { bold: false });
+    expect(getCell(model, "B2")!.style).toEqual({ bold: false });
+    setCellContent(model, "B2", "updated b2");
+    expect(getCellContent(model, "B2")).toBe("updated b2");
+
+    paste(model, "C2");
+
+    expect(getCellContent(model, "C2")).toBe("b2");
+    expect(getCell(model, "C2")!.style).toEqual({ bold: true });
+  });
+
+  test("can paste old content/style even if content/style is altered after cut", () => {
+    const model = new Model();
+    setCellContent(model, "B2", "b2");
+    expect(getCellContent(model, "B2")).toBe("b2");
+    setStyle(model, "B2", { bold: true });
+    expect(getCell(model, "B2")!.style).toEqual({ bold: true });
+
+    cut(model, "B2");
+
+    setStyle(model, "B2", { bold: false });
+    expect(getCell(model, "B2")!.style).toEqual({ bold: false });
+    setCellContent(model, "B2", "updated b2");
+    expect(getCellContent(model, "B2")).toBe("updated b2");
+
+    paste(model, "C2");
+
+    expect(getCellContent(model, "C2")).toBe("b2");
+    expect(getCell(model, "C2")!.style).toEqual({ bold: true });
+  });
+
+  test("cut zone gets cleared on paste if content/style is altered after cut", () => {
+    const model = new Model();
+    setCellContent(model, "B2", "b2");
+    expect(getCellContent(model, "B2")).toBe("b2");
+    setStyle(model, "B2", { bold: true });
+    expect(getCell(model, "B2")!.style).toEqual({ bold: true });
+
+    cut(model, "B2");
+
+    setStyle(model, "B2", { bold: false });
+    expect(getCell(model, "B2")!.style).toEqual({ bold: false });
+    setCellContent(model, "B2", "updated b2");
+    expect(getCellContent(model, "B2")).toBe("updated b2");
+
+    paste(model, "C2");
+
+    expect(getCell(model, "B2")).toEqual(undefined);
+    expect(getCellContent(model, "C2")).toBe("b2");
+    expect(getCell(model, "C2")!.style).toEqual({ bold: true });
+  });
+
+  test("copying external content & paste-format on a cell will not paste content", async () => {
+    const model = new Model();
+    const clipboardData = new MockClipboardData();
+    clipboardData.setData(ClipboardMIMEType.PlainText, "Excalibur");
+
+    const content = clipboardData.getData(ClipboardMIMEType.PlainText);
+    pasteFromOSClipboard(model, "C2", content);
+    expect(getCellContent(model, "C2")).toBe(content);
+    pasteFromOSClipboard(model, "C3", content, "onlyFormat");
+    expect(getCellContent(model, "C3")).toBe("");
+  });
+
+  test("Internal copy followed by OS copy should not bring paste format from internal copy", async () => {
+    const model = new Model();
+    setCellContent(model, "A1", "a1");
+    setStyle(model, "A1", { fillColor: "#FA0000" });
+    copy(model, "A1");
+    expect(getStyle(model, "A1")).toEqual({ fillColor: "#FA0000" });
```

Comment:
This is useless no?

Reply:
Main objective behind this test case was to check an edge case with "onlyFormat". Once we copy something internally, & followed we make an external copy, it was observed that upon special pasting format, it was bringing the format of first internal copy, which is obviously wrong. Hence style and content is set beforehand. Although your point does stay that pasteFromOSClipboard method will disregard the above anyway, But then I don't really can figure out how to recreate this issue in test case.

Reply:
I get your point. The test should be in grid.ts and use the copy/paste function of the grid then. That way you mimick real life behaviour. 

Doing it here is useless, as using `pasteFromOSClipboard` will totally ignore the clipboard state. Doing the test in the grid actually test wheter you call `paste` or `pasteFromOS`, and thus whetehr you paste the internal style or not.